pinot-common/src/main/java/org/apache/pinot/common/utils/helix/IdealStateGroupCommit.java

```diff
@@ -124,34 +124,27 @@ public IdealState commit(HelixManager helixManager, String resourceName,
           Entry first = queue._pending.poll();
           processed.add(first);
           String mergedResourceName = first._resourceName;
```

Comment:
Not quite related to the fix, but I think 
`Queue queue = getQueue(resourceName);`
so queue should already only has resourceName? 
Why do we pull mergedResourceName again?

Reply:
Not quite --
```java
private Queue getQueue(String resourceName) {
  return _queues[(resourceName.hashCode() & Integer.MAX_VALUE) % _queues.length];
}
```
So just it's a bit likely the queue only has 1 resource but not necessarily